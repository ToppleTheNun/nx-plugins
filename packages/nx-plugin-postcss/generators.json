{
  "$schema": "http://json-schema.org/schema",
  "name": "nx-plugin-postcss",
  "version": "0.0.1",
  "generators": {
    "nx-plugin-postcss": {
      "factory": "./src/generators/nx-plugin-postcss/generator",
      "schema": "./src/generators/nx-plugin-postcss/schema.json",
      "description": "nx-plugin-postcss generator"
    }
  }
}
