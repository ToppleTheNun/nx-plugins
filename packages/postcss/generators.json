{
  "$schema": "http://json-schema.org/schema",
  "name": "nx-plugin-postcss",
  "version": "0.0.1",
  "generators": {
    "init": {
      "factory": "./src/generators/init/init",
      "schema": "./src/generators/init/schema.json",
      "description": "Add PostCSS configuration to the workspace (add dependencies)",
      "aliases": ["ng-add"],
      "hidden": true
    },
    "configuration": {
      "factory": "./src/generators/configuration/configuration",
      "schema": "./src/generators/configuration/schema.json",
      "description": "Add PostCSS configuration to a UI library or an application",
      "hidden": false
    }
  }
}
