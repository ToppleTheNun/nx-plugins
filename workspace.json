{
  "version": 2,
  "projects": {
    "e2e-postcss": "e2e/postcss",
    "e2e-utils": "e2e/utils",
    "postcss": "packages/postcss"
  }
}
