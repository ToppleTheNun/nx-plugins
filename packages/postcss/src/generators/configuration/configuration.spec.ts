import {
  addDependenciesToPackageJson,
  readJson,
  readProjectConfiguration,
  Tree,
} from "@nrwl/devkit";
import { createTreeWithEmptyWorkspace } from "@nrwl/devkit/testing";
import { libraryGenerator } from "@nrwl/workspace/generators";
import configurationGenerator from "./configuration";

describe("@topplethenun/nx-plugin-postcss:configuration", () => {
  let tree: Tree;

  beforeEach(async () => {
    tree = createTreeWithEmptyWorkspace();
    await libraryGenerator(tree, {
      name: "test-ui-lib",
      standaloneConfig: false,
    });
  });

  describe("dependencies for package.json", () => {
    it("adds postcss if not already present", async () => {
      const existing = "existing";
      const existingVersion = "1.0.0";
      addDependenciesToPackageJson(
        tree,
        { [existing]: existingVersion },
        { [existing]: existingVersion }
      );

      await configurationGenerator(tree, { project: "test-ui-lib" });

      const packageJson = readJson(tree, "package.json");
      expect(packageJson).toMatchSnapshot();

      expect(packageJson.devDependencies["postcss"]).toEqual("^8.0.0");
      expect(packageJson.devDependencies["postcss-preset-env"]).toEqual(
        "^7.0.0"
      );
    });

    it("does not add postcss if already present", async () => {
      const existing = "existing";
      const existingVersion = "1.0.0";
      addDependenciesToPackageJson(
        tree,
        { [existing]: existingVersion },
        { postcss: "7", [existing]: existingVersion }
      );

      await configurationGenerator(tree, { project: "test-ui-lib" });

      const packageJson = readJson(tree, "package.json");
      expect(packageJson).toMatchSnapshot();

      expect(packageJson.devDependencies["postcss"]).toEqual("7");
      expect(packageJson.devDependencies["postcss-preset-env"]).toEqual(
        "^7.0.0"
      );
    });

    it("does not add postcss-preset-env if already present", async () => {
      const existing = "existing";
      const existingVersion = "1.0.0";
      addDependenciesToPackageJson(
        tree,
        { [existing]: existingVersion },
        { "postcss-preset-env": "7", [existing]: existingVersion }
      );

      await configurationGenerator(tree, { project: "test-ui-lib" });

      const packageJson = readJson(tree, "package.json");
      expect(packageJson).toMatchSnapshot();

      expect(packageJson.devDependencies["postcss"]).toEqual("^8.0.0");
      expect(packageJson.devDependencies["postcss-preset-env"]).toEqual("7");
    });
  });

  it("should generate files", async () => {
    await configurationGenerator(tree, { project: "test-ui-lib" });

    expect(tree.exists("libs/test-ui-lib/postcss.config.js")).toBeTruthy();
  });

  it("should update workspace file", async () => {
    await configurationGenerator(tree, {
      project: "test-ui-lib",
    });
    const project = readProjectConfiguration(tree, "test-ui-lib");

    expect(project.targets.build).toEqual({
      executor: "@topplethenun/nx-plugin-postcss:package",
      outputs: ["{options.outputPath}"],
      options: {
        outputPath: "dist/libs/test-ui-lib/dist",
        packageJson: "libs/test-ui-lib/package.json",
        main: "libs/test-ui-lib/src/index.css",
        assets: [
          "libs/test-ui-lib/*.md",
          { glob: "**/*", input: "libs/test-ui-lib/src", output: "./src" },
        ],
      },
    });
  });
});
