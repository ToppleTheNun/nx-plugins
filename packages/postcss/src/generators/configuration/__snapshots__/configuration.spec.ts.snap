// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@topplethenun/nx-plugin-postcss:configuration dependencies for package.json adds postcss if not already present 1`] = `
Object {
  "dependencies": Object {
    "existing": "1.0.0",
    "tslib": "^2.0.0",
  },
  "devDependencies": Object {
    "@nrwl/eslint-plugin-nx": "13.4.1",
    "@nrwl/jest": "13.4.1",
    "@nrwl/linter": "13.4.1",
    "@types/jest": "27.0.2",
    "@typescript-eslint/eslint-plugin": "~5.3.0",
    "@typescript-eslint/parser": "~5.3.0",
    "eslint": "8.2.0",
    "eslint-config-prettier": "8.1.0",
    "existing": "1.0.0",
    "jest": "27.2.3",
    "postcss": "^8.0.0",
    "postcss-load-config": "^3.0.0",
    "postcss-preset-env": "^7.0.0",
    "ts-jest": "27.0.5",
  },
  "name": "test-name",
}
`;

exports[`@topplethenun/nx-plugin-postcss:configuration dependencies for package.json does not add postcss if already present 1`] = `
Object {
  "dependencies": Object {
    "existing": "1.0.0",
    "tslib": "^2.0.0",
  },
  "devDependencies": Object {
    "@nrwl/eslint-plugin-nx": "13.4.1",
    "@nrwl/jest": "13.4.1",
    "@nrwl/linter": "13.4.1",
    "@types/jest": "27.0.2",
    "@typescript-eslint/eslint-plugin": "~5.3.0",
    "@typescript-eslint/parser": "~5.3.0",
    "eslint": "8.2.0",
    "eslint-config-prettier": "8.1.0",
    "existing": "1.0.0",
    "jest": "27.2.3",
    "postcss": "7",
    "postcss-load-config": "^3.0.0",
    "postcss-preset-env": "^7.0.0",
    "ts-jest": "27.0.5",
  },
  "name": "test-name",
}
`;

exports[`@topplethenun/nx-plugin-postcss:configuration dependencies for package.json does not add postcss-preset-env if already present 1`] = `
Object {
  "dependencies": Object {
    "existing": "1.0.0",
    "tslib": "^2.0.0",
  },
  "devDependencies": Object {
    "@nrwl/eslint-plugin-nx": "13.4.1",
    "@nrwl/jest": "13.4.1",
    "@nrwl/linter": "13.4.1",
    "@types/jest": "27.0.2",
    "@typescript-eslint/eslint-plugin": "~5.3.0",
    "@typescript-eslint/parser": "~5.3.0",
    "eslint": "8.2.0",
    "eslint-config-prettier": "8.1.0",
    "existing": "1.0.0",
    "jest": "27.2.3",
    "postcss": "^8.0.0",
    "postcss-load-config": "^3.0.0",
    "postcss-preset-env": "7",
    "ts-jest": "27.0.5",
  },
  "name": "test-name",
}
`;
